vs_utils/features/nnscore.py

```diff
@@ -0,0 +1,735 @@
+# HERE].
+"""
+
+__author__ = "Bharath Ramsundar"
+__license__ = "GNU General Public License"
+
+import textwrap
+import math
+import os
+import sys
+import textwrap
+import glob
+import cPickle
+import numpy as np
+import itertools
+from vs_utils.features import Featurizer
+from vs_utils.features.nnscore_utils import MathFunctions
+from vs_utils.features.nnscore_pdb import PDB
+from vs_utils.features.nnscore_utils import Point
+
+ELECTROSTATIC_JOULE_PER_MOL = 138.94238460104697e4 # units?
+# O-H distance is 0.96 A, N-H is 1.01 A. See
+# http://www.science.uwaterloo.ca/~cchieh/cact/c120/bondel.html
+H_BOND_DIST = 1.3 # angstroms
+H_BOND_ANGLE = 40 # degrees
+# If atoms are < 2.5 A apart, we count it as a close contact
+CLOSE_CONTACT_CUTOFF = 2.5
+# If receptor and ligand atoms are > 4 A apart, we consider them
+# unable to interact with simple electrostatics.
+CONTACT_CUTOFF = 4 # angstroms
+# "PI-Stacking Interactions ALIVE AND WELL IN PROTEINS" says
+# distance of 7.5 A is good cutoff. This seems really big to me,
+# except that pi-pi interactions (parallel) are actually usually
+# off centered. Interesting paper.  Note that adenine and
+# tryptophan count as two aromatic rings. So, for example, an
+# interaction between these two, if positioned correctly, could
+# count for 4 pi-pi interactions.
+PI_PI_CUTOFF = 7.5
+# Cation-pi interaction cutoff based on
+# "Cation-pi interactions in structural biology."
+CATION_PI_CUTOFF = 6.0
+# 4  is good cutoff for salt bridges according to
+# "Close-Range Electrostatic Interactions in Proteins",
+# but looking at complexes, I decided to go with 5.5 A
+SALT_BRIDGE_CUTOFF = 5.5
+# This is perhaps controversial. I noticed that often a pi-cation
+# interaction or other pi interaction was only slightly off, but
+# looking at the structure, it was clearly supposed to be a pi-cation
+# interaction. I've decided then to artificially expand the radius of
+# each pi ring. Think of this as adding in a VDW radius, or
+# accounting for poor crystal-structure resolution, or whatever you
+# want to justify it.
+PI_PADDING = 0.75
+
+
+def hashtable_entry_add_one(hashtable, key, toadd = 1):
+  # note that dictionaries (hashtables) are passed by reference in python
+  if hashtable.has_key(key):
+    hashtable[key] = hashtable[key] + toadd
+  else:
+    hashtable[key] = toadd
+
+
+def extend_list_by_dictionaries(lst, dictionaries):
+  newlist = lst[:]
+  for dictionary in dictionaries:
+    # first, sort the dictionary by the key
+    keys = dictionary.keys()
+    keys.sort()
+
+    # now make a list of the values
+    vals = []
+    for key in keys:
+      vals.append(dictionary[key])
+
+    # now append vals to the list
+    newlist.extend(vals)
+
+  # return the extended list
+  return newlist
+
+def center(string, length):
+  while len(string) < length:
+    string = " " + string
+    if len(string) < length:
+      string = string + " "
+  return string
+
+
+class binana:
```

Comment:
Uppercased it. I'm not sure about whether it should be separate file or not. The author packaged binana within his nnscore script, which is why I've retained the nnscore naming convention. I'm leaning toward leaving it as is unless there's a good for separating it out.